src/migration/ember-addon/steps/update-addon-package-json.js

```diff
@@ -24,7 +24,7 @@ function updateDependencies(packageJson) {
   const packagesToInstall = ['@embroider/addon-shim'];
 
   packagesToInstall.forEach((packageName) => {
-    const version = decideVersion(packageName, dependencies);
+    const version = decideVersion(packageName, options);
```

Comment:
**bugfix**

We should pass `options` (to be precise, `augmentedOptions`) because it includes the combined list of dependencies from `dependencies` and `devDependencies`.